chall06/krusthol.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/python3
+import sys
+
+READ_ERR = "Can't read file"
+FIT_ERR = "Not enough space in the given shelves"
+
+def print_proceed(many_args, std_input, index, message):
+	if many_args:
+		print(f"{sys.argv[index]}:")
+	if std_input:
+		print(f"{sys.argv[0]}: stdin: {message}")
+	else:
+		print(f"{sys.argv[0]}: {sys.argv[index]}: {message}")
+	return (index + 1)
+
+def book_fitter(index):
+	many_args = len(sys.argv) > 2
+	std_input = len(sys.argv) == 1
+	books = []
+	if std_input:
+		lines = []
+		while True:
+			try:
+				std_line = input()
+				if std_line == '':
+					break
+				lines.append(std_line)
+			except EOFError:
+				break
+		if not lines:
+			return (print_proceed(many_args, std_input, index, READ_ERR))
+		shelves = lines[0].split()
+		lines.pop(0)
+		for line in lines:
+			new_book = line.split()[0]
+			books.append(new_book)
+	elif not std_input:
+		try:
+			with open(sys.argv[index]) as open_file:
+				shelves = open_file.readline().split()		
+				line = open_file.readline()
```

Comment:
not sure it has exactly the same behaviors

but I think readline().split() may be written readlines()